model/file_info.go

```diff
@@ -6,72 +6,169 @@ package model
 import (
 	"bytes"
 	"encoding/json"
+	"image"
 	"image/gif"
 	"io"
 	"mime"
 	"path/filepath"
+	"strings"
 )
 
 type FileInfo struct {
-	Filename        string `json:"filename"`
-	Size            int    `json:"size"`
+	Id              string `json:"id"`
+	UserId          string `json:"user_id"`
+	PostId          string `json:"post_id,omitempty"`
```

Comment:
Posts do actually have Files as well. It's a two-way relationship right now since I convinced Corey to let me store the FileIds on the Post.

You got me thinking about it a bit to remember why we need it, but we need the FileInfo.PostId field because we can't query SQL based on the Post.FileIds since it's serialized as JSON. Without it, we would need multiple database calls to get the FileInfos for a post and it would be much more complicated to find a FileInfo's Post to do a permissions check.

As for multiple posts sharing a file, I don't think we're planning on supporting that any time soon, but we'd need to use a join table if we did. That's not a huge deal to eventually add [since it could probably be done with pure SQL](https://gist.github.com/hmhealey/db1fe8eb65b9b95f65657ea51e359cb8). I don't think we should add it now though since it would just add extra complexity to support something we don't need right now.